src/client/opamCommands.ml

```diff
@@ -750,7 +750,30 @@ let list ?(force_search=false) cli =
     let results =
       OpamListCommand.filter ~base:all st filter
     in
-    if not no_depexts && not silent then
+    let uses_depexts = function
+      | OpamListCommand.Any
+      | OpamListCommand.Installed
+      | OpamListCommand.Root
+      | OpamListCommand.Compiler
+      | OpamListCommand.Pinned
+      | OpamListCommand.Latests_only
+      | OpamListCommand.Pattern _
+      | OpamListCommand.Atoms _
+      | OpamListCommand.Flag _
+      | OpamListCommand.Tag _
+      | OpamListCommand.From_repository _
+      | OpamListCommand.Owns_file _
+        -> false
+      | OpamListCommand.Available
+      | OpamListCommand.Installable
+      | OpamListCommand.Depends_on _
+      | OpamListCommand.Required_by _
+      | OpamListCommand.Conflicts_with _
+      | OpamListCommand.Coinstallable_with _
+      | OpamListCommand.Solution _
+        -> true
```

Comment:
It should be a function in `OpamListCommand`.

Reply:
done